rnd/autogpt_server/autogpt_server/data/graph.py

```diff
@@ -0,0 +1,166 @@
+import asyncio
+import json
+
+from prisma.models import AgentGraph, AgentNode, AgentNodeLink, AgentNodeExecution
+
+from autogpt_server.data.db import BaseDbModel
+
+
+class Edge(BaseDbModel):
+    input_node: str
+    input_name: str
+    output_node: str
+    output_name: str
+
+    @staticmethod
+    def from_db(edge: AgentNodeLink):
+        return Edge(
+            id=edge.id,
+            input_node=edge.agentNodeInputId,
+            input_name=edge.inputName,
+            output_node=edge.agentNodeOutputId,
+            output_name=edge.outputName,
+        )
+
+
+class Node(BaseDbModel):
+    block_name: str
```

Comment:
I think this should reference a block by id, what if block name changes?

Reply:
The name is the id. The old id (UUID) will always change whenever the app starts up making the value inconsistent, hence I removed it. Unless we want to add UUID on each class block implementation (and that point, same applies, what if the UUID change?).

Reply:
If we identify a block using a value that can change we might want to add another immutable value and use it as reference instead.
As I see it now we cannot change `name` because we'll lose reference to blocks. But imagine we have `ProcessStraemAPIBlockV2` and we cannot change its name (I made typo intentionally :) ), I think it's easier to have easy way to change the visible name, than make some wrappers in future that map `block_name`->`visible name`.

Reply:
It still should be possible to use block name when constructing graphs so it's easy to do in code/json I mean that only for internal references.

Reply:
I see, Good point on typos, I've added id() on Block interface.